lib/async.js

```diff
@@ -75,6 +77,7 @@ module.exports = function resolve(x, options, callback) {
     var extensions = opts.extensions || ['.js'];
     var basedir = opts.basedir || path.dirname(caller());
     var parent = opts.filename || basedir;
+    var env = opts.ignoreExportsField === false ? ['require', 'node'] : [];
```

Comment:
I think that there should be an option to specify the `env` array.

Reply:
See https://github.com/browserify/resolve/pull/224#issuecomment-678476270 - I'm intentionally not exposing that just yet.